Purchases/Networking/HTTPClient.swift

```diff
@@ -224,9 +224,12 @@ private extension HTTPClient {
             }
         }
 
+        var maybeNetworkError = maybeNetworkError
         if DNSChecker.isBlockedAPIError(maybeNetworkError),
-            let resolvedHost = DNSChecker.blockedHostFromError(maybeNetworkError) {
-            Logger.error(NetworkStrings.blocked_network(newHost: resolvedHost))
+            let blockedError = DNSChecker.errorWithBlockedHostFromError(maybeNetworkError) {
```

Comment:
Following up to my other comment, I think it would make sense to create a whole new error here with our domain and code that wraps `maybeNetworkError`, then `isAPIBlockedError` can check against that.

Reply:
I updated this. What do you think of the new Error type in the DNSChecker?